lib/util/DocumentExtractor.js

```diff
@@ -42,18 +42,24 @@ const extractors = ([
     methods: {
       extractFromRequest: request => {
         const documents = [];
-        if (request.input.body && Object.keys(request.input.body).length) {
+        if ( request.input.body
+          && request.input.body.ids
+          && Object.keys(request.input.body.ids).length
+        ) {
           for (let it = 0; it < request.input.body.ids.length; it++) {
             documents.push({ _id: request.input.body.ids[it] });
           }
-        } else if (request.input.args.ids) {
+        }
+        else if (request.input.args.ids) {
           if (Array.isArray(request.input.args.ids)) {
             documents.push({ _id: request.input.args.ids });
             return documents;
           }
-          const ids = request.input.args.ids.split(',');
-          for (let it = 0; it < ids.length; it++) {
-            documents.push({ _id: ids[it] });
+          if (typeof request.input.args.ids === 'string') {
```

Comment:
:+1: 

you should also throw a meaningful error if ids is neither an array nor a string (`api.assert.invalid_type` seems suited)